packages/protocol/contracts-0.8/stability/FeeCurrencyAdapter.sol

```diff
@@ -0,0 +1,168 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity >=0.8.7 <0.8.20;
+
+import "@openzeppelin/contracts8/access/Ownable.sol";
+import "@openzeppelin/contracts8/token/ERC20/IERC20.sol";
+
+import "../../contracts/common/CalledByVm.sol";
+import "../../contracts/common/Initializable.sol";
+import "../../contracts/common/interfaces/ICeloVersionedContract.sol";
+import "../../contracts/common/FixidityLib.sol";
+import "../../contracts/stability/interfaces/ISortedOracles.sol";
+import "./interfaces/IFeeCurrency.sol";
+import "./interfaces/IDecimals.sol";
+
+contract FeeCurrencyAdapter is Initializable, CalledByVm {
+  IFeeCurrency public adaptedToken;
+
+  uint96 public digitDifference;
+
+  uint256 public debited;
+
+  string public name;
+  string public symbol;
+
+  event GasFeesDebited(address indexed debitedFrom, uint256 debitedAmount);
```

Comment:
how much gas will this cost us? the ERC20 will be already emitting events here `Tranfer()` events